lib/charms/observability_libs/v1/cert_handler.py

```diff
@@ -161,6 +322,34 @@ def __init__(
             self.charm.on[self.certificates_relation_name].relation_broken,  # pyright: ignore
             self._on_certificates_relation_broken,
         )
+        self.framework.observe(
+            self.charm.on.upgrade_charm,  # pyright: ignore
+            self._on_upgrade_charm,
+        )
+
+    def _on_upgrade_charm(self, _):
+        self._migrate_vault()
+
+    def _migrate_vault(self):
+        peer_backend = _RelationVaultBackend(self.charm, relation_name="peers")
+
+        if self._check_juju_supports_secrets():
+            # we are on recent juju
+            if self.vault.retrieve():
```

Comment:
`retrieve` is a tuple and always evaluates to true.
But we probably don't need `retrieve` anyway and should be using `get_value`?

Reply:
`retrieve()` is a `Dict[str,str]` and will evaluate to False if it's empty.